nx/include/switch/kernel/semaphore.h

```diff
@@ -0,0 +1,22 @@
+/**
+ * @file semaphore.h
+ * @brief Thread synchronization based on Mutex.
+ * @author Kevoot
+ * @copyright libnx Authors
+ */
+#pragma once
+
+#include "mutex.h"
+#include "condvar.h"
+
+typedef struct Semaphore
+{
+    CondVar condvar;
+    Mutex mutex;
+    u64 count;
+} Semaphore;
+
+void semaphoreInit(Semaphore *, u64);
```

Comment:
I think this should have Doxygen documentation too, like the other synchronization primitives. Also, the parameter names are missing in the prototypes. `Semaphore*` should probably be `s` and the `u64` should be `initial_count`.